runtime/include/atomics/intrinsics/chpl-atomics.h

```diff
@@ -28,19 +28,34 @@
   #include <intrinsics.h>
 #endif
 
+//
+// SIZE_ALIGN_TYPE:  Declare a version of a type aligned to at least its size.
+//
+// C11 syntax does not support doing this to typedefs.  However,
+// all the compilers we support with the intrinsics version of atomics
+// also support the gcc alignment attribute syntax used here.
```

Comment:
What do you think of dropping the C11/compiler support info from this comment? I think it's useful information and could go in the PR message, but it doesn't seem especially helpful for context in this file (and hopefully in the future if the compiler supports C11/_Alignas, we'll be using the C11 atomics implementation instead of the intrinsics one.)